python/graphstorm/eval/eval_func.py

```diff
@@ -695,17 +702,65 @@ def compute_mae(pred, labels):
     return th.mean(diff).cpu().item()
 
 def compute_mrr(ranking):
-    """ Get link prediction mrr metrics
+    """ Get link prediction Mean Reciprocal Rank (MRR) metrics
 
-        Parameters
-        ----------
-        ranking:
-            ranking of each positive edge
+    Parameters
+    ----------
+    ranking: torch.Tensor
+        ranking of each positive edge
 
-        Returns
-        -------
-        link prediction mrr metrics: tensor
+    Returns
+    -------
+    th.Tensor
+        link prediction mrr metrics
     """
-    logs = th.div(1.0, ranking)
-    metrics = th.tensor(th.div(th.sum(logs),len(logs)))
+    reciprocal_ranks = th.div(1.0, ranking)
+    metrics = th.tensor(th.div(th.sum(reciprocal_ranks), len(reciprocal_ranks)))
     return metrics
+
+def compute_amri(ranking: th.Tensor, candidate_sizes: th.Tensor) -> th.Tensor:
+    """Computes the Adjusted Mean Rank Index (AMRI) for the given ranking and candidate sizes.
+
+    AMRI is a metric that evaluates the performance of link prediction models by considering both
+    the rank of the correct candidate and the number of candidates considered. It is calculated as:
```

Comment:
We include the positive example so it would be all candidates (we use `num_negative+1`). From the paper:

> To compute a rank in LP and EA evaluation, we are given a list of scores `S = [β1, . . . , βC ]` for each test instance with `|C| = C`, where C is a set of candidates. We denote the score of the ”true” entity as α, and its position in the decreasingly sorted list as `r = rank(S, α)`.